invenio_curations/assets/semantic-ui/js/invenio_curations/deposit/DepositBox.js

```diff
@@ -167,9 +167,15 @@ DepositBoxComponent.defaultProps = {
   groupsEnabled: false,
 };
 
-const mapStateToProps = (state) => ({
-  record: state.deposit.record,
-});
+function mapStateToProps(state) {
+  if ("parent" in state.deposit.record) {
+    return {
+      record: state.deposit.record,
+    };
+  }
+  return {};
+}
```

Comment:
With v12 (12.0.5) or v13 (13.0.0b1.dev3)., there are no issues locally with the old code. Could you elaborate on what the exact issue is?

Reply:
funny. i tried to reproduce it, was not possible it is working now. it cost me a few hours to find out what the problem was and how to solve it and now it seems it is gone.

so the problem was that by creating a new record and saving it for the first time the `record` object didn't have the `links` and `parent` attributes in it. the only solution i found was the fix in the above commit. i had no idea why the fix should be necessary in the first place. i thought maybe there was some sort of updates somewhere i didn't could locate which introduces some sort of edge cases which are hit. not knowing in detail what was going on was also the reason why the commit messages looked like that. 

i will remove that commit.

Reply:
It probably also depends on how we want to tackle the underlying issue. Your solution works as well. We just do not get the form record state until it has been saved at least once - which happens rather quickly usually. If we keep this solution, we should add a comment about why this is necessary.

Otherwise we should modify the visibility check for the draft button from
`{(record.is_draft === null || permissions.can_manage) && (`
to
`{(record?.ui || permissions.can_manage) &&`
so it shows up if the draft has been saved once.

Reply:
My understanding was that currently, the first check is for visiting `/uploads/new` and the second check is for visiting `/uploads/<recid>`. `can_manage` will be false on a new draft but true for an existing draft, when visiting the page again. `record.parent` and `record.ui` will also be set when visiting the draft page again or when saving a new draft.

Reply:
i've tackled this problem differently in #10 , inspired by the way that the `ShareDraftComponent` in Invenio-App-RDM is hooked up ([v12.0.5](https://github.com/inveniosoftware/invenio-app-rdm/blob/v12.0.5/invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/deposit/ShareDraftButton.js#L58-L65))
[here is the commit](https://github.com/tu-graz-library/invenio-curations/pull/10/commits/8c864d0b6b3a4f4ad671cde07a9772f0ce4697e0) with more details